Lib/fontTools/varLib/__init__.py

```diff
@@ -208,6 +212,62 @@ def _add_avar(font, axes):
         assert +1.0 not in curve or curve[+1.0] == +1.0
         # curve.update({-1.0: -1.0, 0.0: 0.0, 1.0: 1.0})
 
+    if mappings:
+        interesting = True
+
+        hiddenAxes = [axis for axis in axes.values() if axis.hidden]
+
+        inputLocations = [
+            {
+                axes[name].tag: models.normalizeValue(v, vals_triples[axes[name].tag])
+                for name, v in mapping.inputLocation.items()
+            }
+            for mapping in mappings
+        ]
+        outputLocations = [
+            {
+                axes[name].tag: models.normalizeValue(v, vals_triples[axes[name].tag])
+                for name, v in mapping.outputLocation.items()
+            }
+            for mapping in mappings
+        ]
+
+        # If base-master is missing, insert it at zero location.
+        if not any(all(v == 0 for k, v in loc.items()) for loc in inputLocations):
+            inputLocations.insert(0, {})
+            outputLocations.insert(0, {})
```

Comment:
you're only checking if base master is missing from inputLocations but then add to outputLocations unconditionally, should you check each one independently?

Reply:
The two are parallel lists at this point. Ie. the outputLocations is the value corresponding to the input location. So when we insert in one we have to insert into other. Since there's no data for the default master available, the value would be empty dictionary.